File: src/main/java/de/tum/cit/aet/artemis/core/dto/PasskeyDto.java
package de.tum.cit.aet.artemis.core.dto;

import java.time.Instant;

/**
 * DTO for passkey information.
 *
 * @param credentialId encoded as base64url
 */
public record PasskeyDto(String credentialId, String label, long signatureCount, Instant created, Instant lastUsed) {
}

File: src/main/webapp/app/core/user/settings/passkey-settings/dto/passkey.dto.ts
export interface PasskeyDTO {
    /**
     * The credential ID of the passkey, encoded in Base64url format.
     */
    credentialId: string;
    label: string;

    /**
     * ISO 8601 date string
     */
    created: string;

    /**
     * ISO 8601 date string
     */
    lastUsed: string;

    /**
     * Super admin approval is required for privileged features (e.g. administrator features)
     */
    isSuperAdminApproved: boolean;
}
